include/swift/AST/ASTMangler.h

```diff
@@ -348,7 +348,8 @@ class ASTMangler : public Mangler {
                                       GenericSignature signature,
                                       ResilienceExpansion expansion);
 
-  std::string mangleTypeForDebugger(Type decl, GenericSignature sig);
+  std::string mangleTypeForDebugger(Type decl, GenericSignature sig,
+                                    bool RespectOriginallyDefinedIn = false);
```

Comment:
This is only called in three places. How about instead of adding a parameter here, you just change the three callers to set RespectOriginallyDefinedIn to true or false as appropriate